### PR TITLE
TST: adapt GPKG tests with undefined CRS

### PR DESCRIPTION
**Pull request by martinfleis**
xref #1975

This should make CI green (with the exception of dev) again. I would keep the #1975 open because we need to resolve it properly I think.

### CODE REVIEW

**Comment by martinfleis** [comment]
`ModuleNotFoundError: No module named 'pandas'` ehm. Something is broken.

File: geopandas/io/tests/test_file.py [hunk 1]
```diff
@@ -275,6 +273,22 @@ def test_append_file(tmpdir, df_nybb, df_null, driver, ext):
     assert_geodataframe_equal(df, expected, check_less_precise=True)
 
 
+@pytest.mark.xfail
+def test_gpkg_empty_crs(tmpdir):
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file.py:277]: We could maybe directly parametrize this test for the different drivers?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for figuring this out!

The strange CI error is also happening on master (in the nightly build: https://github.com/geopandas/geopandas/runs/2957679373?check_suite_focus=true)

**Comment by jorisvandenbossche** [comment]
https://github.com/geopandas/geopandas/pull/1987 to fix the CI issue

File: geopandas/io/tests/test_file.py (hunk 1 above)
Comment (martinfleis) [reply geopandas/io/tests/test_file.py]: Done.

File: geopandas/io/tests/test_file.py [hunk 2]
```diff
@@ -275,6 +273,28 @@ def test_append_file(tmpdir, df_nybb, df_null, driver, ext):
...
+            "geometry": [Point(0, 0), Point(1, 1), Point(2, 2)],
+        },
+    )
+
+    df.to_file(tempfilename, driver=driver)
+    result = read_file(tempfilename)
+
+    if driver == "GeoJSON":
+        # geojson by default assumes epsg:4326
+        result.crs = None
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file.py:293]:
```suggestion
        df.crs = "EPSG:4326"
```

(then it actually tests that it is set like that, although that might already be covered elsewhere)

File: geopandas/io/tests/test_file.py [hunk 3]
```diff
@@ -275,6 +273,28 @@ def test_append_file(tmpdir, df_nybb, df_null, driver, ext):
     assert_geodataframe_equal(df, expected, check_less_precise=True)
 
 
+@pytest.mark.xfail(strict=False)
+@pytest.mark.parametrize("driver,ext", driver_ext_pairs)
+def test_gpkg_empty_crs(tmpdir, driver, ext):
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file.py:278]:
```suggestion
def test_empty_crs(tmpdir, driver, ext):
```

**Comment by jorisvandenbossche** [comment]
Does the empty crs test only fail for GPKG? If that's the case, maybe only xfail that?

**Comment by martinfleis** [comment]
I think this is ready to be merged.

**Comment by jorisvandenbossche** [comment]
Thanks!

In the dev build, it seems there is an additional pandas related failure in addition to the matplotlib color issue

**Pull request merged**